lib/plug/static.ex

```diff
@@ -117,49 +117,48 @@ defmodule Plug.Static do
   end
 
   def init(opts) do
-    at = Keyword.fetch!(opts, :at)
-    from = Keyword.fetch!(opts, :from)
-    gzip? = Keyword.get(opts, :gzip, false)
-    brotli? = Keyword.get(opts, :brotli, false)
-    only = Keyword.get(opts, :only, [])
-    prefix = Keyword.get(opts, :only_matching, [])
-
-    qs_cache = Keyword.get(opts, :cache_control_for_vsn_requests, "public, max-age=31536000")
-    et_cache = Keyword.get(opts, :cache_control_for_etags, "public")
-    et_generation = Keyword.get(opts, :etag_generation, nil)
-    headers = Keyword.get(opts, :headers, %{})
-
     from =
-      case from do
+      case from = Keyword.fetch!(opts, :from) do
         {_, _} -> from
         _ when is_atom(from) -> {from, "priv/static"}
         _ when is_binary(from) -> from
         _ -> raise ArgumentError, ":from must be an atom, a binary or a tuple"
       end
 
-    {Plug.Router.Utils.split(at), from, gzip?, brotli?, qs_cache, et_cache, et_generation, only, prefix, headers}
+    %{
+      gzip?: Keyword.get(opts, :gzip, false),
+      brotli?: Keyword.get(opts, :brotli, false),
+      only: Keyword.get(opts, :only, []),
+      prefix: Keyword.get(opts, :only_matching, []),
+      qs_cache: Keyword.get(opts, :cache_control_for_vsn_requests, "public, max-age=31536000"),
+      et_cache: Keyword.get(opts, :cache_control_for_etags, "public"),
+      et_generation: Keyword.get(opts, :etag_generation, nil),
+      headers: Keyword.get(opts, :headers, %{}),
+      from: from,
+      at: opts |> Keyword.fetch!(:at) |> Plug.Router.Utils.split()
+    }
   end
 
-  def call(conn = %Conn{method: meth}, {at, from, gzip?, brotli?, qs_cache, et_cache, et_generation, only, prefix, headers})
+  def call(conn = %Conn{method: meth}, options)
       when meth in @allowed_methods do
-    segments = subset(at, conn.path_info)
+    segments = subset(options.at, conn.path_info)
```

Comment:
Same here: please rely on the pattern match at head instead of `options.at`. It is a more efficient way to extract multiple values on perfomance sensitive code.